### ISSUE TITLE
facing error  if (fer?.code !== 'EISDIR') { 

### ISSUE THREAD

**Issue by abhishekdhull21** [isaacs/node-mkdirp#39]
Hey facing an issue when building on AWS

/codebuild/output/src306813184/src/node_modules/mkdirp-promise/node_modules/mkdirp/dist/cjs/mkdirp-manual.js:17

 if (fer?.code !== 'EISDIR') {
SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)

**Comment by mukesh-nuo** [comment]
We are also facing same issue:
```
error [request-id:null]: uncaughtException: Unexpected token '.'
/home/ubuntu/app/node_modules/mkdirp/dist/cjs/mkdirp-manual.js:17
            if (fer?.code !== 'EISDIR') {
                    ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/ubuntu/app/node_modules/mkdirp/dist/cjs/index.js:4:28)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/ubuntu/app/node_modules/mkdirp-promise/lib/index.js:3:16

**Comment by alexanderglueck** [comment]
`mkdirp-promise` pulls in `mkdirp` without locking it to a specific version (`"mkdirp": "*",`).
You should be able to (temporarily) fix the issue by adding a specific version of `mkdirp` to your `package.json` dependencies.

**Comment by isaacs** [comment]
Yeah, relying on `*` is unsafe at any speed.  SemVer major bumps are _for_ breaking changes!

That said, `engines` does allow v10, because I forgot to bump it at mkdirp@v2, so this is a legit bug.  Fixed on v2.1.1.

**Comment by isaacs** [comment]
The fix is 3a097bd245618f36fe943394dd40cca7fcfa37a8
